manim/mobject/probability.py

```diff
@@ -157,23 +161,82 @@ def __getitem__(self, index):
 
 
 class BarChart(VGroup):
+    """This is a class for Bar Charts.
+
+    Parameters
+    ----------
+    values
+        The values for the bar chart.
+    height
+        The height of the axes.
+    width
+        The width of the axes.
+    n_ticks
+        Number of ticks.
+    tick_width
+        Width of the ticks.
+    label_y_axis
+        Y axis label
+    y_axis_label_height
+        Height of the label.
+    max_value
+        Maximum value of the data.
+    bar_colors
+        The colors of the bars.
+    bar_fill_opacity
+        The opacity of the bars.
+    bar_stroke_width
+        The stroke width of the bars.
+    bar_names
+        The names of each bar.
+    bar_label_scale_val
+        The label size.
+
+    Examples
+    --------
+    .. manim:: BarChartExample
+
+        class BarChartExample(Scene):
+            def construct(self):
+                lg = [54, 23, 47, 48, 40, 64, 112, 87]
+                versions = [
+                    "v0.1.0",
+                    "v0.1.1",
+                    "v0.2.0",
+                    "v0.3.0",
+                    "v0.4.0",
+                    "v0.5.0",
+                    "v0.6.0",
+                    "v0.7.0",
+                ]
+                colors = ["#003f5c", "#58508d", "#bc5090", "#ff6361", "#ffa600"]
+                bar = BarChart(
+                    lg,
+                    max_value=max(lg),
+                    bar_colors=colors,
+                    bar_names=versions,
+                    bar_label_scale_val=0.3,
+                )
+                self.add(bar)
+    """
+
     def __init__(
         self,
-        values,
-        height=4,
-        width=6,
-        n_ticks=4,
-        tick_width=0.2,
-        label_y_axis=True,
-        y_axis_label_height=0.25,
-        max_value=1,
-        bar_colors=[BLUE, YELLOW],
-        bar_fill_opacity=0.8,
-        bar_stroke_width=3,
-        bar_names=[],
-        bar_label_scale_val=0.75,
+        values: Iterable[float],
+        height: int = 4,
```

Comment:
Please be careful with adding type hints. Are you sure that height can only be int and not e.g. float?
My friendly advice (only meant in the best way, not meant as an accusation): Please only add type hints when you are 100% sure that any other option would not be possible.
Also, I remember there were some really complicated functions in `space_ops` that got type hints, where I was not sure if they were right. As tyephints is not the easiest programming topic, I would also suggest that we revert some not 100% save changes in `space_ops`, because it is deftly better to have no typehints then wrong typehints.

Reply:
Hey, it just makes more sense to have height as an `int` rather than a `float`. Also for `space_ops` I double checked the typehints and removed the ones I'm not sure of. I also put some links for better understanding of some functions over there and how it works and the way we implemented it. If something is still wrong, let me know I can clarify or remove it 😄

Reply:
> just makes more sense to have height as an int rather than a float 
But it would be also *possible* to init it with let's say width = 6.5 , no?
Typehints should include all possibilities.

Reply:
width = 6
![image](https://user-images.githubusercontent.com/34582615/120931750-951c5c00-c710-11eb-8710-491879c13deb.png)
width = 6.5
![image](https://user-images.githubusercontent.com/34582615/120931754-9c436a00-c710-11eb-9bba-c87de2b5836b.png)
width = 7
![image](https://user-images.githubusercontent.com/34582615/120931775-b8470b80-c710-11eb-9472-fd619d1cd204.png)

Is there any difference between 6 and 6.5 ? I found a slight one for 6 and 7,

Reply:
yes, the with is differnt, see my screenshot:
![image](https://user-images.githubusercontent.com/44469195/120934798-0c95bf80-c700-11eb-8e36-c616dc612c78.png)